src/components/form/group-controls/checkbox-group/docs.mdx

```diff
@@ -0,0 +1,243 @@
+---
+componentName: CheckboxGroup
+---
+
+## `isRequired` and `isOptional`
+
+### With `legend` and `isRequired`
+
+<CheckboxGroup
+	isRequired
+	legend="Legend of the group"
+	options={[
+		{
+			label: 'Label of control #1',
+		},
+		{
+			label: 'Label of control #2',
+		},
+		{
+			label: 'Label of control #3',
+		},
+	]}
+/>
```

Comment:
thought: It would be great to see a more compositional approach here:

```jsx
<CheckboxGroup ... >
	<CheckboxField ... />
	<CheckboxField ... />
	<CheckboxField ... />
</CheckboxGroup>
```

Reply:
I don't think we should use a compositional approach here because there are quite a few things unique to `CheckboxField`s when inside of a `CheckboxGroup`:

- Cannot show `helperText` when `CheckboxGroup`'s `layout` is `'horizontal'`
- Label should have `'regular'` font weight when `layout` is `'vertical'`, `'semibold'` otherwise
- The `name` prop is set based on `CheckboxGroup`'s `name` prop (so all checkboxes should have the same `name`)
- Do not want to show errors for individual checkboxes, so `errors` must be omitted

We can handle all of this automatically with less code if we don't take a compositional approach. If you're looking for a compositional approach for styling reasons, I think we could allow a `className` prop in `CheckboxGroupOption`? Otherwise, I plan to make a `FieldSet` component outside of this PR, and that would enable creating a custom `CheckboxGroup` with a compositional approach if absolutely needed.